tf-modules/s3-credentials/main.tf

```diff
@@ -0,0 +1,227 @@
+}
+
+resource "aws_security_group" "s3_credentials_lambda" {
+  count = (var.vpc_id != null) ? 1 : 0
+
+  vpc_id = var.vpc_id
+
+  egress {
+    from_port   = 0
+    to_port     = 0
+    protocol    = "-1"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+  tags = var.tags
+}
+
+resource "aws_lambda_function" "s3_credentials" {
+  function_name    = "${var.prefix}-s3-credentials-endpoint"
+  filename         = local.lambda_source_file
+  source_code_hash = filebase64sha256(local.lambda_source_file)
+  handler          = "index.handler"
+  role             = aws_iam_role.s3_credentials_lambda.arn
+  runtime          = "nodejs12.x"
+  timeout          = 10
+  memory_size      = 320
+
+  vpc_config {
+    subnet_ids         = var.lambda_subnet_ids
+    security_group_ids = (var.vpc_id != null) ? [aws_security_group.s3_credentials_lambda[0].id] : null
+  }
+
+  environment {
+    variables = {
+      DISTRIBUTION_ENDPOINT          = var.external_api_endpoint
+      DISTRIBUTION_REDIRECT_ENDPOINT = "${var.external_api_endpoint}redirect"
+      public_buckets                 = join(",", var.public_buckets)
+      EARTHDATA_BASE_URL             = var.urs_url
+      EARTHDATA_CLIENT_ID            = var.urs_client_id
+      EARTHDATA_CLIENT_PASSWORD      = var.urs_client_password
+      AccessTokensTable              = aws_dynamodb_table.access_tokens.id
+      STSCredentialsLambda           = var.sts_credentials_lambda_function_arn
+    }
+  }
+  tags = var.tags
+}
+
+data "aws_region" "current" {}
+
+resource "aws_lambda_permission" "lambda_permission" {
+  action        = "lambda:InvokeFunction"
+  function_name = aws_lambda_function.s3_credentials.function_name
+  principal     = "apigateway.amazonaws.com"
+
+  # The /*/*/* part allows invocation from any stage, method and resource path
+  # within API Gateway REST API.
+  source_arn = "arn:aws:execute-api:${data.aws_region.current.name}:${data.aws_caller_identity.current.account_id}:${var.rest_api_id}/*/*/*"
+}
+
+# GET /s3credentials
+resource "aws_api_gateway_resource" "s3_credentials" {
+  rest_api_id = var.rest_api_id
+  parent_id = var.rest_api_root_resource_id
+  path_part   = "s3credentials"
+}
+
+resource "aws_api_gateway_method" "s3_credentials" {
+  rest_api_id   = var.rest_api_id
+  resource_id   = aws_api_gateway_resource.s3_credentials.id
+  http_method   = "GET"
+  authorization = "NONE"
+}
+
+resource "aws_api_gateway_integration" "s3_credentials" {
+  rest_api_id             = var.rest_api_id
+  resource_id             = aws_api_gateway_resource.s3_credentials.id
+  http_method             = aws_api_gateway_method.s3_credentials.http_method
+  integration_http_method = "POST"
+  type                    = "AWS_PROXY"
+  uri                     = aws_lambda_function.s3_credentials.invoke_arn
+}
+
+# GET /redirect
+resource "aws_api_gateway_resource" "s3_credentials_redirect" {
+  rest_api_id = var.rest_api_id
+  parent_id = var.rest_api_root_resource_id
+  path_part   = "redirect"
```

Comment:
indentation is not very clean here

Reply:
1f608a2ac8788a4747cad792c7e62261e93bd56c